### ISSUE TITLE
Problem decoding source (pyflakes E) (Spyder 4)

### ISSUE THREAD

**Issue by OverLordGoldDragon** [spyder-ide/spyder#11154]
Image below; occurred seemingly randomly when editing code in a separate Spyder window (via "new window"); no clue how to reproduce. As is seen, commenting out the offending line doesn't help; what does: (1) commenting out entire file code; (2) causing syntax error elsewhere in code (but as soon as either is undone, error returns).

Also, when I un-commented that first line, Spyder froze and I had to restart it (after which the error's gone). I like the new features, but hope these are fixed soon as they're rather disruptive.

<hr>

![image](https://user-images.githubusercontent.com/16495490/71302877-fb50da00-23c9-11ea-9f4b-a7854c997655.png)

**Comment by ccordoba12** [comment]
> Image below; occurred seemingly randomly when editing code in a separate Spyder window; no clue how to reproduce

You need to provide us with a reproducible use case. Otherwise, we'll be unable to solve this.

> I like the new features, but hope these are fixed soon as they're rather disruptive.

I understand your frustration, but this kind of error reports are also frustrating for us because it's more like a complaint than a proper bug report.

**Comment by OverLordGoldDragon** [comment]
>this kind of error reports are also frustrating for us because it's more like a complaint than a proper bug report.

Agreed, but again they appear to simply come and go - it's rather a lose-lose situation. If I manage to find reproducible steps, will update.

**Comment by texadactyl** [comment]
@OverLordGoldDragon What you have shown are imports without how they are used or where they came from.  Can you reproduce the symptoms that you described in a smaller program with fewer external import dependencies? A source file that can be compiled by Python 3 or analyzed by pylint by other people would facilitate investigation.

**Comment by ccordoba12** [comment]
@OverLordGoldDragon, I was also thinking about this yesterday, and it could be related to using an encoding different than `utf-8` in the files you opened in Spyder's editor.

Is that the case with the file you shown above?

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 I haven't pasted anything into the `.py` files which I typed from scratch, and haven't tweaked encoding options. Regardless, how would I view what encoding a `.py` file uses?

@texadactyl `model_utils` imports from `keras` or `tensorflow.keras` depending on the `os.environ["TF_KERAS"]` flag - I'll see if I can narrow it down next time it occurs

**Comment by ccordoba12** [comment]
> I haven't pasted anything into the .py files, which I typed from scratch

Did you write them entirely in Spyder?

> Regardless, how would I view what encoding a .py file uses?

It's shown in the status bar:

![Selección_069](https://user-images.githubusercontent.com/365293/71416310-db215500-262d-11ea-8fb5-48a0528bcfc7.png)

In this case, the file has the UTF-8 encoding.

**Comment by OverLordGoldDragon** [comment]
> Did you write them entirely in Spyder?

Cannot guarantee I never pasted a single character from elsewhere, no - but vast majority is indeed typed in Spyder

And... bingo, you may be onto something:

![image](https://user-images.githubusercontent.com/16495490/71418004-ccfd2400-2681-11ea-8396-5570a1008c8a.png)

I haven't had this specific problem recur ever since, but in case encoding is important for some other bugs or general stability - how do I swap to UTF-8?

**Comment by ccordoba12** [comment]
You need to add this comment

```python
# -*- coding: utf-8 -*-
```

to the top of your files (i.e. it needs to be the first line in them).

**Comment by cyf203** [comment]
@ccordoba12 I think even add ```# -*- coding: utf-8 -*-``` doesn't help. I had the same issue. When I create a new file, the error icon occurs exactly like above. @OverLordGoldDragon The way I solve this problem is to close the file and reopen the same script. Then the icon is gone.

**Comment by goanpeca** [comment]
I do not think we are changing the encoding on the fly with just adding that line, that is why the reopen is necessary. Other IDEs provide the option to change the encoding with a menu entry. We could also provide thus by making the status bar item clickable (which would display enconding options with a menu/dropdown)

**Comment by rabbitfoot1** [comment]
Had this problem today. Restarting spyder fixed it.

**Comment by OverLordGoldDragon** [comment]
To notify, I've never had this problem since - though I've been using `# -*- coding: utf-8 -*-` atop each file.

**Comment by andfoy** [comment]
@OverLordGoldDragon, could you please provide us with a minimum reprducible file that presents this issue?

**Comment by OverLordGoldDragon** [comment]
@andfoy Unsure, the behavior is spontaneous - I was just editing a file and it popped up. There is no file that has this marking upon simply opening it. Maybe there's a 'pattern' though - it did happen after linter marked a Syntax error which I resolved; I'll notify if I find anything.

**Comment by vahid-sb** [comment]
I just faced the same issue, googled it, and ended up in here. Since I could not find a solution here, I thought to close and reopen the file, and that solved the problem. Also, while I faced this error, all other errors in the script were ignored by the IDE, but warnings, such as docstring missing, were shown. 
I write all my scripts in Spyder, and always use default settings, encoding UTF-8, etc. And this was the first time I bumped into this error. Interestingly enough, this was the first time in Spyder 4 I was writing a class in a separate file. Not sure if it is significant, but at least we now know one easy solution to get rid of the error.

**Comment by frankcoumans** [comment]
I have the same issue 1-2 times per week since creating a new environment with spyder 4. I don't recall seeing it in the half year I've worked with spyder 3. All my files start with _# -*- coding: utf-8 -*-_ and I've changed only the appearance (to pydev). Pyflakes locks up, red cross stays where it is, also if the line is now empty. Restarting Spyder to resolve works. I also run keras. I have not yet observed a way to trigger it, so I won't be able to create a use case.

**Comment by ccordoba12** [comment]
@OverLordGoldDragon, have you seen this after PR #12771?

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 No, but I've seen it only once since 4.12 or 4.11 anyway, so doubt my install is a useful reference. If 4.14 isn't coming soon, perhaps we could suggest users to try the PR fix.

**Comment by ccordoba12** [comment]
Agreed. @frankcoumans, please check PR #12771 for a possible fix for this.

**Comment by frankcoumans** [comment]
Thank you! Applied the fix. Will let you know if I see it again.

**Comment by frankcoumans** [comment]
@ccordoba12 I havent't seen pyflakes lock up since applying PR #12771 about a week ago.

Thanks again!

**Comment by ccordoba12** [comment]
@frankcoumans, thanks for the confirmation! Let's close this one then.

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 It popped up again. And I still have #12771 applied.

**Comment by ccordoba12** [comment]
Please open a new issue about that and try to give us a reproducible case or video for us to reproduce it.

**Comment by frankcoumans** [comment]
I also occasionally have the issue, but Pyflakes doesn't stay locked on an error anymore; the red x now disappears when the offending line is removed. However, Pyflakes just stops checking the code for a while and then resumes operation (without reopening the file, but after running the code in debug mode). It's not really been a problem, and it's so rare that I have no clue how to reproduce.

**Comment by OverLordGoldDragon** [comment]
@ccordoba12 If I can reproduce, sure, but similar to @frankcoumans, this is very rare and minimally problematic as resolution is trivial.

**Comment by ccordoba12** [comment]
Ok, that's really good to know! Please keep us posted after 4.1.4 is released to see if things stay like that or regress to the previous behavior.

**Comment by Ainsleydmb** [comment]
Hello, 

I am having this same issue and it happens with this code every times on both my computers. I have tried the solutions here and none have worked, so I am wondering if there actually is an error in the code? 

<img width="629" alt="Screenshot 2021-08-20 at 10 24 11" src="https://user-images.githubusercontent.com/36626519/130212044-3a3d86fd-3257-42b5-8378-30dbb65af3fe.png">

**Comment by ccordoba12** [comment]
Hey @Ainsleydmb, the code you posted above is not valid Python, it's bash code. So there's no wonder our linter is giving this error for you.
